src/listfunc.c

```diff
@@ -266,8 +266,9 @@ Obj             FuncAPPEND_LIST_INTR (
         GROW_STRING(list1, len1 + len2);
         SET_LEN_STRING(list1, len1 + len2);
         CLEAR_FILTS_LIST(list1);
-        // copy data, including terminating zero byte
-        memcpy(CHARS_STRING(list1) + len1, CONST_CHARS_STRING(list2), len2 + 1);
+        // copy data, but not terminating byte, in case list1==list2
+        memcpy(CHARS_STRING(list1) + len1, CONST_CHARS_STRING(list2), len2);
+        CSTR_STRING(list1)[len1 + len2] = '\0';
```

Comment:
Good catch. I guess alternatively we could use `SyMemmove`?

Reply:
(I am not saying we should, just mumbling :)

Reply:
That's probably better, done.